Source/WebCore/rendering/MotionPath.cpp

```diff
@@ -62,6 +62,15 @@ std::optional<MotionPathData> MotionPath::motionPathDataForRenderer(const Render
         }
         return std::nullopt;
     }
+    if (is<ShapePathOperation>(pathOperation)) {
+        if (auto* container = renderer.containingBlock()) {
+            auto& shapePathOperation = downcast<ShapePathOperation>(*pathOperation);
+            data.containingBlockBoundingRect = snapRectToDevicePixelsIfNeeded(container->referenceBoxRect(shapePathOperation.referenceBox()), downcast<RenderLayerModelObject>(renderer));
+            data.offsetFromContainingBlock = offsetFromContainer(renderer, *container);
+            return data;
+        }
+        return std::nullopt;
+    }
```

Comment:
I wonder if we can cleanly share code somehow with `BoxPathOperation`. Not a big deal if we can't though.

Reply:
Yeah I'm planning on sharing them once referenceBox() goes on the root PathOperation in https://github.com/WebKit/WebKit/pull/17140.